cmd/traffic/cmd/manager/internal/mutator/watcher_test.go

```diff
@@ -0,0 +1,214 @@
+package mutator
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"golang.org/x/sync/errgroup"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/datawire/k8sapi/pkg/k8sapi"
+	mock_kubernetes "github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/manager/internal/mutator/mocks"
+	"github.com/telepresenceio/telepresence/v2/pkg/agentconfig"
+)
+
+//go:generate go run github.com/golang/mock/mockgen -package=mock_kubernetes -destination=mocks/k8s_interface_mock.go k8s.io/client-go/kubernetes Interface
+//go:generate go run github.com/golang/mock/mockgen -package=mock_kubernetes -destination=mocks/k8s_corev1_mock.go k8s.io/client-go/kubernetes/typed/core/v1 CoreV1Interface
+//go:generate go run github.com/golang/mock/mockgen -package=mock_kubernetes -destination=mocks/k8s_configmap_mock.go k8s.io/client-go/kubernetes/typed/core/v1 ConfigMapInterface
+type suiteConfigWatcher struct {
+	suite.Suite
+
+	ctrl *gomock.Controller
+
+	kubeApiMock      *mock_kubernetes.MockInterface
+	coreV1ApiMock    *mock_kubernetes.MockCoreV1Interface
+	configMapApiMock *mock_kubernetes.MockConfigMapInterface
```

Comment:
Why aren't we using `k8s.io/client-go/kubernetes/fake` here like we do in the `agent_injector_tests`? Seems to me like that would put things more to the test than a simple mock.

Reply:
Didn't know about this one ! Let me take a look.

Reply:
I was migrating to the fake client, but I noticed that it doesn't allow me to emulate a slow API for example, so I can't migrate my last test :(